controller/search.go

```diff
@@ -249,11 +254,40 @@ func (c *SearchController) WorkitemsCSV(ctx *app.WorkitemsCSVSearchContext) erro
 }
 
 // getWorkItemsByFilterExpression retrieves Work Items for a given expression and parameters
-func getWorkItemsByFilterExpression(ctx context.Context, db application.DB, filterExpression string, filterParentexists *bool, offset *int, limit *int) ([]workitem.WorkItem, error) {
+func getWorkItemsByFilterExpression(ctx context.Context, db application.DB, filterExpression string, filterParentexists *bool, offset *int, limit *int) ([]workitem.WorkItem, link.WorkItemLinkList, link.AncestorList, error) {
 	var result []workitem.WorkItem
+	var childLinks link.WorkItemLinkList
+	var parents link.AncestorList
 	err := application.Transactional(db, func(appl application.Application) error {
 		var err error
-		result, _, _, _, err = appl.SearchItems().Filter(ctx, filterExpression, filterParentexists, offset, limit)
+		// add tree option to query if not already present
+		var reqJSON interface{}
+		err = json.Unmarshal([]byte(filterExpression), &reqJSON)
+		if err != nil {
+			return errs.Errorf("error unmarshalling query expression for CSV filtering: %s", filterExpression)
+		}
+		reqMap := reqJSON.(map[string]interface{})
+		isTreeView := false
+		opts, ok := reqMap["$OPTS"]
+		if ok {
```

Comment:
@michaelkleinhenz We could simplify this code
```go
		var reqMap map[string]interface{}
		err = json.Unmarshal([]byte(filterExpression), &reqMap)
		if err != nil {
			return errs.Errorf("error unmarshalling query expression for CSV filtering: %s", filterExpression)
		}
		if _, ok := reqMap["$OPTS"]; !ok {
			reqMap["$OPTS"] = make(map[string]interface{})
		}
		// Set "tree-view" to true. We always want tree-view to be enabled
		(reqMap["$OPTS"].(map[string]interface{}))["tree-view"] = true
```

What do you think?

Reply:
Looks good, replaced it in af9300e.

Reply:
Thank you